### DISCUSSION TITLE
Running code by using an item

### DISCUSSION THREAD

**Discussion by lazar-the-coder** [peter-kish/gloot#121] [Q&A]
Trying to make a medkit, so I want code to run that increases the players health when they activate the item.
Is there functionality for this, if so, how would i do this?

**Comment by peter-kish** [comment]
There's a few ways to do something like that:
1. Use the `CtrlInventory.inventory_item_activated(item)` signal (seems like it's missing from the documentation for [`CtrlInventory`](https://github.com/peter-kish/gloot/blob/master/docs/ctrl_inventory.md) - I'll have to fix that). All `CtrlInventory*` classes should have it defined. It gets emitted when an item in the inventory is double-clicked.
2. Use the `CtrlInventory.get_selected_inventory_item()` to get the selected inventory when the user hits a key or clicks on a button.

Once you have the `InventoryItem` that needs to be activated, it's just a matter of healing the player and removing the item from the inventory. This can be done by:
1. calling `Inventory.remove_item(item)` to remove the item, or
2. decreasing the item stack size (in case of a `InventoryStacked` or `InventoryGridStacked`) :
```gdscript
set_item_stack_size(item, get_item_stack_size() - 1)
```
Note: In case of item stacks, decreasing the stack size to 0 will not remove the item automatically. You'll probably need to call `remove_item(item)` in that case. Edit: Opened an [issue](https://github.com/peter-kish/gloot/issues/122 ) for this. Hoping to implement it in v2.3.6.

**Reply by lazar-the-coder** [reply]
Thank you for the response, I was actually wondering about the "making the activation run the script that heals them" part, but this is still helpful as the way I was doing the part you mentioned was streamlined with your advice. I also figured the script part out so all's well.

**Comment by peter-kish** [comment]
One more thing:
If you just want to pick any medkit from the inventory, you can also do `Inventory.get_item_by_id(<medkit prototype ID>)` which just returns the first item it can find with the given prototype ID.

**Comment by peter-kish** [comment]
[v2.3.6](https://github.com/peter-kish/gloot/releases/tag/v2.3.6) is out. Setting the item stack size to 0 via set_item_stack_size will now automatically remove it from the inventory and queue it for deletion.
